pylint/pyreverse/inspector.py

```diff
@@ -32,7 +40,11 @@ def _astroid_wrapper(func, modname):
     return None
 
 
-def interfaces(node, herited=True, handler_func=_iface_hdlr):
+def interfaces(
+    node: nodes.ClassDef,
+    herited: bool = True,
+    handler_func: Callable[[nodes.NodeNG | Any], bool] = _iface_hdlr,
+) -> Iterator[Any] | None:
```

Comment:
I think this should be typed as a `Generator` using something like `Generator[Any, None, None]`.

Reply:
Good catch, thank you.